UnitTests/GTMSessionFetcherFetchingTest.m

```diff
@@ -3419,7 +3423,30 @@ - (instancetype)initWithUserAgentBlock:(UserAgentBlock)userAgentBlock {
   return self;
 }
 
+- (instancetype)initWithBlockedTimeout:(NSUInteger)seconds
+                        userAgentBlock:(nonnull UserAgentBlock)userAgentBlock {
+  if (!seconds) {
+    [NSException raise:NSInvalidArgumentException format:@"You must use a nonzero time to wait"];
+  }
+  if (self = [self initWithUserAgentBlock:userAgentBlock]) {
+    _waitSeconds = seconds;
+    _semaphore = dispatch_semaphore_create(0);
+  }
+  return self;
+}
+
+- (void)unblock {
+  NSAssert(_waitSeconds, @"This was not a blocked UAProvider");
+  dispatch_semaphore_signal(_semaphore);
+  _waitSeconds = 0;
+}
+
 - (NSString *)userAgent {
+  if (_waitSeconds) {
+    intptr_t waitResult = dispatch_semaphore_wait(
+        _semaphore, dispatch_time(DISPATCH_TIME_NOW, _waitSeconds * NSEC_PER_SEC));
+    XCTAssertEqual(0, waitResult);
```

Comment:
May be useful for debugging to add a `,@"Timed out after %d seconds", __waitSeconds);`

Reply:
Done, and also did that to the TestAuthorizer support.